windows/src/desktop/setup/UfrmRunDesktop.pas

```diff
@@ -1032,25 +1032,41 @@ procedure TfrmRunDesktop.Status(const Text: WideString = '');
 
 procedure TfrmRunDesktop.GetDefaultSettings; // I2651
 begin
+  // Due to the way the TUtilKeymanOptionEntry.Save works it only
+  // saves the values that are different from the defaults.  So we
+  // need to set the defaults here. Two locations for the defaults is
+  // not ideal, but leave for now.
   FStartWithWindows := True; // I2607
   FCheckForUpdates := True;  // I2609
   FAutomaticallyReportUsage := True;
 end;
 
 procedure TfrmRunDesktop.GetSavedSettings;
+var
+  CheckForUpdatesFound: Boolean;
```

Comment:
It might be clearer to name this `CheckForUpdatesKeyFound` - as it is right now I find it easy to confuse with `CheckForUpdates`